### PR TITLE
shellcheck corrections

### PR DESCRIPTION
**Pull request by Yarboa**
shellcheck tests are broken
fixing script based on shell check recommendations

sample https://github.com/containers/initoverlayfs/actions/runs/6946556735

### CODE REVIEW

**Review by ericcurtin — APPROVED** [review]
I think this is ok, just a question, did we test this?

I worry about things like:

`| "$decompressor" |`

do they work like we intend, with the lack of CI in this project at present, we have to manually test every change until we get proper CI, that executes this stuff e2e.

**Comment by ericcurtin** [comment]
Thanks for looking into this @Yarboa I think this is our first green PR!

File: scripts/build.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ fs="erofs"
 extract_initrd_into_initoverlayfs() {
   sudo mkdir -p "$DIR_TO_DUMP_INITRAMFS"
 
-  file_type=$(file /boot/initramfs-$release.img)
+  file_type=$(file /boot/initramfs-"$release.img")
```
[the remark this answers is not in the document]
Comment (Yarboa) [reply scripts/build.sh:24]: Good catch, thanks

**Comment by Yarboa** [comment]
Moving to draft, 
We will merge this one with automation, good point @ericcurtin

**Comment by Yarboa** [comment]
@ericcurtin 
PR was tested with f39 release

**Comment by ericcurtin** [comment]
Thanks for testing, mergy mergy

**Pull request merged**